### PR TITLE
Add cassandra health indicator and enable probes

### LINKED ISSUE #192: Add liveness endpoint to REST API
The following REST endpoint is provided today (by the Spring framework?)

> http://< host >:< port >/actuator/health

Does that guarantee that the ecChronos application is live (working as expected)? 
Or is a separate endpoint provided by ecChronos needed, to verify that the repair scheduling is working as expected.

It should be possible to use this REST endpoint as an liveness-probe in a k8s cluster.

### PR DESCRIPTION
**Pull request by masokol**
actuator/health/readiness will report readiness of ecchronos.
The above will check CQL and JMX connection towards cassandra.
actuator/health/liveness includes generic springboot liveness.

Closes #192

### CODE REVIEW

**Comment by masokol** [comment]
We might want to include more checks if someone has any ideas what would be good to have.

File: application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/spring/CassandraHealthIndicator.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                    .withJmxConnectionProvider(myJmxConnectionProvider)
+                    .build();
+            JmxProxy jmxProxy = jmxProxyFactory.connect();
+            jmxProxy.close();
+            return true;
+        }
+        catch (Exception e)
+        {
+            LOG.debug("JMX readiness failed", e);
+            LOG.error("JMX readiness failed {}", e.getMessage());
```
Comment (itskarlsson) [inline application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/spring/CassandraHealthIndicator.java:78]: Not so sure about using readiness terminology in our healthcheck. 

"Unable to connect over JMX" - I would like better.

File: application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/spring/CassandraHealthIndicator.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            Session session = myNativeConnectionProvider.getSession();
+            ResultSet result = session.execute(selectFromLocal);
+            if (!result.all().isEmpty())
+            {
+                return true;
+            }
+        }
+        catch (Exception e)
+        {
+            LOG.debug("CQL readiness failed", e);
```
Comment (itskarlsson) [inline application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/spring/CassandraHealthIndicator.java:98]: Not so sure about using readiness terminology in our healthcheck.

"Unable to connect over CQL" - I would like better.

File: application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/spring/CassandraHealthIndicator.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            LOG.error("CQL readiness failed {}", e.getMessage());
```
Comment (itskarlsson) [inline application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/spring/CassandraHealthIndicator.java:99]: Also not a big fan of double printing error messages. Do we need the error message here?

File: application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/spring/CassandraHealthIndicator.java (hunk 1 above)
Comment (itskarlsson) [inline application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/spring/CassandraHealthIndicator.java:78]: Also not a big fan of double printing error messages. Do we need the error message here?

File: application/src/main/resources/application.yml [hunk 4]
```diff
@@ -24,6 +24,13 @@ management:
   endpoint:
     health:
       enabled: true
+      probes:
+        enabled: true
```
Comment (itskarlsson) [inline application/src/main/resources/application.yml:28]: As I understand it, these probes are enabled by default according to (https://www.baeldung.com/spring-liveness-readiness-probes) when using a Kubernetes environment. Why do we enable it for all environments?

Comment (masokol) [reply application/src/main/resources/application.yml:28]: I don't trust automatic detection, "automatic detection" usually doesn't work 100% of the time. Also, i think it's nice to have alive/ready split up.

**Review by itskarlsson — APPROVED** [review]

**Pull request merged**